pkg/processor/runtime/python/test/python_test.go

```diff
@@ -318,17 +325,21 @@ func (suite *TestSuite) TestNonUTF8Headers() {
 	createFunctionOptions := suite.GetDeployOptions("non-utf8-headers",
 		path.Join(suite.GetTestFunctionsDir(), "common", "empty", "python"))
 	createFunctionOptions.FunctionConfig.Spec.Handler = "empty:handler"
+	internalServerErrorStatus := http.StatusInternalServerError
+	okStatus := http.StatusOK
 
 	nonUTF8String := string([]byte{192, 175})
-	internalServerErrorStatus := http.StatusInternalServerError
+
+	createFunctionOptions.FunctionConfig.Spec.Env = []v1.EnvVar{
+		{Name: "NUCLIO_PYTHON_DECODE_EVENTS", Value: "true"},
+	}
 	suite.DeployFunctionAndRequests(createFunctionOptions, []*httpsuite.Request{
 		{
-			RequestMethod: http.MethodPost,
-			RequestBody:   nonUTF8String,
+			RequestMethod:              http.MethodPost,
+			RequestBody:                nonUTF8String,
+			ExpectedResponseStatusCode: &okStatus,
```

Comment:
Why is a non utf-8 request body ok (when decoding is on) but a header is not?
worth an explanation imo

Reply:
adding
```
// event body is []bytes and hence would always be a python bytestring
// and hence no utf8 decoding is applied by msgpack
```